src/org/zaproxy/zap/extension/codedx/ExtensionAlertHttp.java

```diff
@@ -48,11 +53,18 @@ public String getXml(SiteNode site) {
     private String getHTML(Alert alert) {
         // gets HttpMessage request and response data from each alert and removes illegal and special characters
         StringBuilder httpMessage = new StringBuilder();
+        
+        HttpMessage message = alert.getMessage();
+        
+        if (message == null) {
+        	LOGGER.warn(String.format(Constant.messages.getString("codedx.error.httpMessage"), alert.getAlertId()));
```

Comment:
@kingthorin was suggesting `Constant.messages.getString("codedx.error.httpMessage", alert.getAlertId())` which does not require extra formatting by the caller.